ra_aid/project_state.py

```
@@ -0,0 +1,66 @@
+        bool: True if the directory is empty or contains only git files,
+              False otherwise
+
+    Raises:
+        DirectoryNotFoundError: If the specified directory does not exist
+        DirectoryAccessError: If the directory cannot be accessed
+        ProjectStateError: For other unexpected errors
+    """
+    try:
+        path = Path(directory).resolve()
+        if not path.exists():
+            raise DirectoryNotFoundError(f"Directory does not exist: {directory}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)